src/test/validation_chainstatemanager_tests.cpp

```diff
@@ -97,7 +98,13 @@ BOOST_AUTO_TEST_CASE(chainstatemanager)
     exp_tip = c1.m_chain.Tip();
     BOOST_CHECK_EQUAL(validated_tip, exp_tip);
 
-    // Avoid triggering the address sanitizer.
+    // Drop any events from the scheduler to avoid accessing memory that is going to be unloaded
+    m_node.scheduler->stop();
+    threadGroup.interrupt_all();
+    threadGroup.join_all();
+    GetMainSignals().FlushBackgroundCallbacks();
+    GetMainSignals().UnregisterBackgroundSignalScheduler();
```

Comment:
Would just calling `SyncWithValidationInterfaceQueue` here instead of dropping notifications and cleaning everything up work?

If I'm understanding the bug correctly, the problem is a race condition where scheduled notifications from code above are racing to finish before manager.Unload() below. If this is the case, calling Sync should be a better fix because it would remove nondeterminism from the test (notifications non deterministically getting executing or dropped). Also it would avoid having cleanup code in the test that duplicates fixture cleanup code and runs twice

Reply:
Thanks. Done